PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -60,7 +81,15 @@ private static Exception createPayPalNotEnabledError() {
                 "for more information.");
     }
 
-    public void onActivityResumed(BrowserSwitchResult browserSwitchResult, PayPalNativeOnActivityResumedCallback callback) {
-
+    public void onActivityResumed(final BrowserSwitchResult browserSwitchResult, final PayPalNativeOnActivityResumedCallback callback) {
+        if (payPalClient != null) {
+            //it means it was a vault request
+            payPalClient.onBrowserSwitchResult(browserSwitchResult, new PayPalBrowserSwitchResultCallback() {
+                @Override
+                public void onResult(@Nullable PayPalAccountNonce payPalAccountNonce, @Nullable Exception error) {
+                    callback.onResult(payPalAccountNonce, error);
+                }
+            });
+        }
```

Comment:
Same. If `payPalClient` is not null, it means it was a vault request, so we default to the current behaviour in `PayPalClient`

Reply:
I think there could be a more reliable assertion here - I believe there are differences in the deep link URL of the `BrowswerSwitchResult` based on the request type, would it make more sense to assert on that?